### PR TITLE
Chore/improve base react template #trivial

### PR DESCRIPTION
**Pull request by kaisermann**
#### What problem is this solving?

- Adds typings for `css` and `graphql`/`gql` files.
- Removes unnecessary options from `tsconfig.json`
- Add default base component (Index.tsx)
- Move default base test next to the tested file
- Simplified the `.vtexignore` and added the `*.test.*` pattern

#### How to test it?

🤷‍♀️

#### Screenshots or example usage:

🤷‍♀️

#### Describe alternatives you've considered, if any.

🤷‍♀️

#### Related to / Depends on

🤷‍♀️

### CODE REVIEW

File: react/index.tsx [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import React from 'react'
+import type { FunctionComponent } from 'react'
+
+type Props = {
+  name: string
+}
+
+const Index: FunctionComponent<Props> = ({ name }) => {
```
Comment (klzns) [inline react/index.tsx:8]: What do you think of avoiding `index.tsx` because of:

```tsx
import { index } from 'vtex.foobar'
```

**Review by klzns — CHANGES_REQUESTED** [review]

File: react/index.tsx (hunk 1 above)
Comment (kaisermann) [reply react/index.tsx:8]: Will call it `Greeting`

**Review by salesfelipe — APPROVED** [review]

**Review by victorhmp — APPROVED** [review]

File: react/Greeting.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import React from 'react'
+import type { FunctionComponent } from 'react'
+
+type Props = {
+  name: string
+}
+
+const Index: FunctionComponent<Props> = ({ name }) => {
```
Comment (klzns) [inline react/Greeting.tsx:8]:
```suggestion
function Greeting({ name }: Props) {
```

File: react/Greeting.tsx [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  return <div>Hi, {name}</div>
+}
+
+export default Index
```
Comment (klzns) [inline react/Greeting.tsx:12]:
```suggestion
export default Greeting
```

File: react/Greeting.tsx [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import React from 'react'
+import type { FunctionComponent } from 'react'
```
Comment (klzns) [inline react/Greeting.tsx:2]:
```suggestion
```

**Review by klzns — CHANGES_REQUESTED** [review]

**Review by klzns — APPROVED** [review]

**Pull request merged**